### PR TITLE
Refactor nook watcher to use new firestore data layout

### PR DESCRIPTION
**Pull request by ktkization**

### CODE REVIEW

File: webapp/lib/controller.dart [hunk 1]
```diff
@@ -65,11 +69,8 @@ class UserData extends Data {
   UserData(this.displayName, this.email, this.photoUrl);
 }
 
-
-Set<String> projectList;
-Set<String> systemEventsProjects;
 List<model.NeedsReplyData> needsReplyDataList;
-List<model.SystemEventsData> systemEventsDataList;
+Map<String, List<model.SystemEventsData>> systemEventsDataList;
```
Comment (marianamarasoiu) [inline webapp/lib/controller.dart]: I think this should be renamed to be a `systemEventsDataMap`?

File: webapp/lib/platform.dart [hunk 2]
```diff
@@ -57,10 +57,12 @@ bool isUserSignedIn() {
 
 typedef CollectionListener(List<DocSnapshot> changes);
 
-void listenForMetrics(String collectionRoot, CollectionListener listener) {
+void listenForMetrics(String collectionRoot, String collectionType, CollectionListener listener) {
```
Comment (marianamarasoiu) [inline webapp/lib/platform.dart]: I suggest we change this to take a single `collectionPath` as an argument, and leaving the callers to construct the path, by calling, e.g.:
```
listenForMetrics('${PROJECTS[i]}/${NEEDS_REPLY_METRICS_COLLECTION_KEY}/metrics')
```

you should be able to then just pass the `collectionPath` argument to 
```
_firestoreInstance
    .collection(collectionPath)
    ....
```

**Review by marianamarasoiu — APPROVED** [review]
thanks @ktkization just a couple of nits, but look good otherwise!

File: webapp/lib/controller.dart (hunk 1 above)
Comment (ktkization) [reply webapp/lib/controller.dart]: Yes, changed

File: webapp/lib/platform.dart (hunk 2 above)
Comment (ktkization) [reply webapp/lib/platform.dart]: Okay. I didn't know I could do that. Thanks for the tip

**Review by marianamarasoiu — APPROVED** [review]
LGTM!

**Pull request merged**